src/schedule/schedule_dataflow_rewrite.cc

```diff
@@ -186,52 +205,108 @@ Tensor CacheWriteWithReLayout(Schedule sch,
     schedule::PassDownIndex(orig_stage, dom_map, &value_map, true);
     for (IterVar iv : orig_stage->leaf_iter_vars) {
       if (red_axis.count(iv)) continue;
-      args.push_back(value_map.at(iv));
+      args->push_back(value_map.at(iv));
     }
   }
-  Operation cache_op = ComputeOpNode::make(
-      compute->name + "." + scope, compute->tag, new_axis, {body});
-  Tensor cache_tensor = cache_op.output(0);
-  Operation orig_new_op = ComputeOpNode::make(
+  std::string name = compute->name;
+  if (0 != scope.size()) {
+    name += ".";
+  }
+  return ComputeOpNode::make(
+      name + scope, compute->tag, new_axis, body_list);
+}
+
+// Cache write and relayout the data according to loop pattern
+Array<Tensor> CacheWriteWithReLayout(Schedule sch,
```

Comment:
I have spend sometime reading the logic but it does not seems to be what we want. For multiple output cases, instead of constructing two tensors that copies the input, we want to create one batched compute op that copies the two tensors ideally. That is

```python
cachedA, cachedB = compute(old_shape, lamdba index: body1, body2)
```